### ISSUE TITLE
reintroduce slerp for camera interpolation

### ISSUE THREAD

**Issue by alisterburt** [napari/napari-animation#37]
I see that camera angles were previously interpolated using spherical linear interpolation [here](https://github.com/sofroniewn/napari-animation/commit/0fb9af0c2d8626b3797facd14ca96179450bb227#diff-ddb53d972021c265e3350c22a11296b5338bf03014148ad91e692e550bdea6f1) but we've now moved to the type based interpolation which directly interpolates the three Euler angles.

Was this done to simplify the code when extending out to interpolating state, rather than just camera properties? If not, is there another reason I'm missing?

**Comment by sofroniewn** [comment]
Yes, this was done to make interpolation as automated as possible, but there are sometime when we want a custom interpolation, for example we want zoom to be exponentially transformed and camera angle to be spherical linear interpolation. We should probably think of some general mechanism where we can specify some properties where we want custom transforms and then do type based interpolation.

Maybe we can provide a dict of dicts like 
```python
{
'camera.zoom':{'forward': lambda x: np.power(10, x), 'inverse':  np.log10)},
'camera.angles':{'forward': some_function, 'inverse': some_inverse_function},
}
```
and then transform each attribute accordingly. How does this sound?

**Comment by Fifourche** [comment]
Hi ! I'm new to all this (napari-animation and github), but I had some questions related to this issue.

Would this Slerp be reintroduced directly on the ArcballCamera _quaternion attribute, or via the Camera Euler angles ? I guess that there would still be some issues  related to gimbal lock if we're using an Euler angles based view description, right ?

**Comment by alisterburt** [comment]
Hi @Fifourche - thanks for your input!

I think we would likely want to stick to using the `angles` property of the camera as input for the state interpolation because this is what's exposed publicly to the user in napari (as `Viewer.camera.angles`). We get that state directly from the camera [here](https://github.com/napari/napari-animation/blob/main/napari_animation/animation.py#L87-L103).

There shouldn't be any problems with gimbal lock because the angles are [generated from/used to generate quaternions](https://github.com/napari/napari/blob/master/napari/_vispy/vispy_camera.py#L45-L70) for the camera. The underlying rotation matrix/quaternion representation generated from the Euler angles is still correct.

If you'd like to help us out on this but would benefit from help contributing do let me know, happy to help you get up and running!

**Comment by alisterburt** [comment]
closed by #62
